ddtrace/debugging/_probe/model.py

```diff
@@ -107,3 +108,16 @@ class MetricProbeKind(object):
 class MetricProbe(LineProbe):
     kind = attr.ib(type=Optional[str], default=None)
     name = attr.ib(type=Optional[str], default=None)
+
+
+@attr.s
+class TemplateSegment(object):
```

Comment:
What if we had subclasses for the str and the expr segments, each implementing its own evaluation method, to which we pass the locals?